test/lsp/ProtocolTest.h

```diff
@@ -95,8 +95,15 @@ class ProtocolTest {
 
     std::unique_ptr<LSPMessage> readAsync();
 
+    template <typename T>
     void assertDiagnostics(std::vector<std::unique_ptr<LSPMessage>> messages, std::vector<ExpectedDiagnostic> expected);
```

Comment:
This shouldn't be in the header file anymore, right?

If someone tried to call this method on a type that wasn't also one of the two called in ProtocolTest.cc, it would be a linker failure.

Reply:
man I need to get a C++ book asap

Reply:
Hmm seems like it needs the template part in the header file? I get an error without it

Reply:
#7227